File: ios/ReservationApp/ReservationApp/Presentation/Views/Admin/Users/Components/UserStatisticsCardsView.swift
//
//  UserStatisticsCardsView.swift
//  ReservationApp
//
//  Created by Mehmet Ali Kısacık on 13.11.2025.
//

import SwiftUI

struct UserStatisticsCardsView: View {
    let statistics: UserStatistics?
    let isLoading: Bool
    
    var body: some View {
        LazyVGrid(columns: [
            GridItem(.flexible(), spacing: ThemeManager.shared.spacing.md),
            GridItem(.flexible(), spacing: ThemeManager.shared.spacing.md)
        ], spacing: ThemeManager.shared.spacing.md) {
            StatisticsCard(
                title: "Toplam Kullanıcı",
                value: statistics?.totalUsers ?? 0,
                isLoading: isLoading,
                color: AppColors.primaryMain
            )
            
            StatisticsCard(
                title: "Aktif",
                value: statistics?.activeUsers ?? 0,
                isLoading: isLoading,
                color: AppColors.successMain
            )
            
            StatisticsCard(
                title: "Pasif",
                value: statistics?.passiveUsers ?? 0,
                isLoading: isLoading,
                color: AppColors.primaryMain
            )
            
            StatisticsCard(
                title: "Bu Ay Yeni",
                value: statistics?.newThisMonth ?? 0,
                isLoading: isLoading,
                color: AppColors.primaryMain
            )
        }
    }
}

struct StatisticsCard: View {
    let title: String
    let value: Int
    let isLoading: Bool
    var color: Color = AppColors.primaryMain
    
    var body: some View {
        VStack(alignment: .leading, spacing: ThemeManager.shared.spacing.xs) {
            Text(title)
                .font(AppTypography.body2())
                .foregroundColor(AppColors.textSecondary)
            
            if isLoading {
                ProgressView()
                    .frame(height: 32)
            } else {
                Text("\(value)")
                    .font(AppTypography.h4())
                    .fontWeight(.semibold)
                    .foregroundColor(color)
            }
        }
        .frame(maxWidth: .infinity, alignment: .leading)
        .padding(ThemeManager.shared.spacing.lg)
        .background(AppColors.backgroundPaper)
        .cornerRadius(ThemeManager.shared.borderRadius.card.value)
        .shadow(color: Color.black.opacity(0.05), radius: 4, x: 0, y: 2)
    }
}

File: android/app/src/main/java/com/reservationapp/ui/admin/users/UserStatisticsCardsView.kt
package com.reservationapp.ui.admin.users

import androidx.compose.foundation.layout.*
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.People
import androidx.compose.material.icons.filled.PersonAdd
import androidx.compose.material.icons.filled.TrendingUp
import androidx.compose.material3.*
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import com.reservationapp.core.ui.theme.*
import com.reservationapp.domain.model.UserStatistics

@Composable
fun UserStatisticsCardsView(
    statistics: UserStatistics?,
    isLoading: Boolean,
    modifier: Modifier = Modifier
) {
    Column(
        modifier = modifier.fillMaxWidth(),
        verticalArrangement = Arrangement.spacedBy(Spacing.md)
    ) {
        Row(
            modifier = Modifier.fillMaxWidth(),
            horizontalArrangement = Arrangement.spacedBy(Spacing.md)
        ) {
            SummaryCard(
                title = "Toplam Kullanıcı",
                value = statistics?.totalUsers ?: 0,
                isLoading = isLoading,
                icon = Icons.Default.People,
                iconColor = PrimaryMain,
                modifier = Modifier.weight(1f)
            )
            SummaryCard(
                title = "Aktif Kullanıcı",
                value = statistics?.activeUsers ?: 0,
                isLoading = isLoading,
                icon = Icons.Default.People,
                iconColor = SuccessMain,
                modifier = Modifier.weight(1f)
            )
        }
        Row(
            modifier = Modifier.fillMaxWidth(),
            horizontalArrangement = Arrangement.spacedBy(Spacing.md)
        ) {
            SummaryCard(
                title = "Pasif Kullanıcı",
                value = statistics?.passiveUsers ?: 0,
                isLoading = isLoading,
                icon = Icons.Default.People,
                iconColor = TextSecondary,
                modifier = Modifier.weight(1f)
            )
            SummaryCard(
                title = "Bu Ay Yeni",
                value = statistics?.newThisMonth ?: 0,
                isLoading = isLoading,
                icon = Icons.Default.PersonAdd,
                iconColor = PrimaryLight,
                modifier = Modifier.weight(1f)
            )
        }
    }
}

@Composable
private fun SummaryCard(
    title: String,
    value: Int,
    isLoading: Boolean,
    icon: androidx.compose.ui.graphics.vector.ImageVector,
    iconColor: Color,
    modifier: Modifier = Modifier
) {
    Card(
        modifier = modifier.fillMaxWidth(),
        colors = CardDefaults.cardColors(
            containerColor = BackgroundPaper
        ),
        elevation = CardDefaults.cardElevation(defaultElevation = 2.dp)
    ) {
        Column(
            modifier = Modifier
                .fillMaxWidth()
                .padding(Spacing.lg),
            verticalArrangement = Arrangement.spacedBy(Spacing.xs)
        ) {
            Text(
                text = title,
                style = MaterialTheme.typography.bodySmall,
                color = TextSecondary
            )

            if (isLoading) {
                CircularProgressIndicator(
                    modifier = Modifier.size(32.dp),
                    color = iconColor
                )
            } else {
                Row(
                    modifier = Modifier.fillMaxWidth(),
                    horizontalArrangement = Arrangement.SpaceBetween,
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    Text(
                        text = "$value",
                        style = MaterialTheme.typography.headlineSmall.copy(
                            fontWeight = FontWeight.SemiBold
                        ),
                        color = iconColor
                    )
                    Icon(
                        imageVector = icon,
                        contentDescription = null,
                        tint = iconColor,
                        modifier = Modifier.size(24.dp)
                    )
                }
            }
        }
    }
}

